spec/models/provider_spec.rb

```diff
@@ -142,27 +137,54 @@
               'address3' => enrichment.address3,
               'address4' => enrichment.address4,
               'postcode' => enrichment.postcode,
-              'region_code' => ProviderEnrichment.region_codes[enrichment.region_code]
+              'region_code' => enrichment.region_code_before_type_cast
             )
           end
 
           context 'enrichment has only region code set for address_info' do
+            london = ProviderEnrichment.region_codes['London']
+            no_region = ProviderEnrichment.region_codes['No region']
+
+            it 'returns address of the provider' do
+              enrichment = build(:provider_enrichment, region_code: no_region,
+              address1: nil,
+              address2: nil,
+              address3: nil,
+              address4: nil,
+              postcode: nil)
+              provider = create(:provider, region_code: london, enrichments: [enrichment])
+
+              expect(provider.address_info).to eq(
+                'address1' => provider.address1,
+                'address2' => provider.address2,
+                'address3' => provider.address3,
+                'address4' => provider.address4,
+                'postcode' => provider.postcode,
+                'region_code' => provider.region_code_before_type_cast
+              )
+            end
+
             it 'returns address of the provider' do
-              london = ProviderEnrichment.region_codes['London']
-              enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
-              'website' => Faker::Internet.url,
-              'region_code' => 'No region',
-              'train_with_us' => Faker::Lorem.sentence.to_s,
-              'train_with_disability' => Faker::Lorem.sentence.to_s })
+              enrichment = build(:provider_enrichment, region_code: no_region, address2: nil,
```

Comment:
How about a new context for this to be clear that we test the way the API works?